internal/slicer/slicer_test.go

```diff
@@ -1478,6 +1478,270 @@ var slicerTests = []slicerTest{{
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Dir(0755, "./"),
+			testutil.Reg(0644, "./file", "foo"),
+			testutil.Lnk(0644, "./symlink", "./file"),
+			testutil.Hlk(0644, "./hardlink", "./symlink"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+					package: test-package
+					slices:
+						myslice:
+							contents:
+								/hardlink:
+								/symlink:
+				`,
+	},
+	filesystem: map[string]string{
+		"/hardlink": "symlink ./file <1>",
+		"/symlink":  "symlink ./file <1>",
+	},
+	manifestPaths: map[string]string{
+		"/symlink":  "symlink ./file <1> {test-package_myslice}",
+		"/hardlink": "symlink ./file <1> {test-package_myslice}",
+	},
+}, {
+	summary: "Hard link identifiers are unique across packages",
+	slices: []setup.SliceKey{
+		{"test-package1", "myslice"},
+		{"test-package2", "myslice"},
+	},
+	pkgs: []*testutil.TestPackage{{
+		Name: "test-package1",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Reg(0644, "./file1", "foo"),
+			testutil.Hlk(0644, "./hardlink1", "./file1"),
+		}),
+	}, {
+		Name: "test-package2",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Reg(0644, "./file2", "foo"),
+			testutil.Hlk(0644, "./hardlink2", "./file2"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package1.yaml": `
+			package: test-package1
+			slices:
+				myslice:
+					contents:
+						/file1:
+						/hardlink1:
+		`,
+		"slices/mydir/test-package2.yaml": `
+			package: test-package2
+			slices:
+				myslice:
+					contents:
+						/file2:
+						/hardlink2:
+		`,
+	},
+	filesystem: map[string]string{
+		"/file1":     "file 0644 2c26b46b <1>",
+		"/hardlink1": "file 0644 2c26b46b <1>",
+		"/file2":     "file 0644 2c26b46b <2>",
+		"/hardlink2": "file 0644 2c26b46b <2>",
+	},
+	manifestPaths: map[string]string{
+		"/file1":     "file 0644 2c26b46b <1> {test-package1_myslice}",
+		"/hardlink1": "file 0644 2c26b46b <1> {test-package1_myslice}",
+		"/file2":     "file 0644 2c26b46b <2> {test-package2_myslice}",
+		"/hardlink2": "file 0644 2c26b46b <2> {test-package2_myslice}",
+	},
+}, {
+	summary: "Mutations for hard links are forbidden",
+	slices: []setup.SliceKey{
+		{"test-package", "myslice"}},
+	pkgs: []*testutil.TestPackage{{
+		Name: "test-package",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Reg(0644, "./file", "foo"),
+			testutil.Hlk(0644, "./hardlink", "./file"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/file:
+						/hardlink: {mutable: true}
+					mutate: |
+						content.write("/hardlink", "new content")
+		`,
+	},
+	error: `slice test-package_myslice: cannot mutate a hard link: /hardlink`,
+}, {
+	summary: "Hard links can be marked as mutable, but not mutated",
```

Comment:
Thanks for this one as well to highlight the issue and not break compatibility unknowingly later.